pkg/controller/machine/controller.go

```diff
@@ -154,6 +154,14 @@ func (r *ReconcileMachine) Reconcile(request reconcile.Request) (reconcile.Resul
 			return reconcile.Result{}, err
 		}
 
+		if m.Status.NodeRef != nil {
+			klog.Infof("Deleting node %q for machine %q", m.Status.NodeRef.Name, m.Name )
+			if err := r.deleteNode(m.Status.NodeRef.Name); err != nil {
```

Comment:
Nit: Why not make `r.deleteNode` take a `context.Context` as first arg instead of creating a new context inside it?